sympy/geometry/util.py

```diff
@@ -684,23 +681,44 @@ def intersection(*entities):
     """
     from .entity import GeometryEntity
     from .point import Point
-
-    if len(entities) <= 1:
-        return []
-
-    # entities may be an immutable tuple
+    pairwise = kwargs.pop('union', False)
+
+    if pairwise == False:
+
+        if len(entities) <= 1:
+            return []
+
+        # entities may be an immutable tuple
+        entities = list(entities)
+        for i, e in enumerate(entities):
+            if not isinstance(e, GeometryEntity):
+                try:
+                    entities[i] = Point(e)
+                except NotImplementedError:
+                    raise ValueError('%s is not a GeometryEntity and cannot be made into Point' % str(e))
+
+        res = entities[0].intersection(entities[1])
+        for entity in entities[2:]:
+            newres = []
+            for x in res:
+                newres.extend(x.intersection(entity))
+            res = newres
+        return res
+
+    ans = []
     entities = list(entities)
     for i, e in enumerate(entities):
         if not isinstance(e, GeometryEntity):
             try:
                 entities[i] = Point(e)
             except NotImplementedError:
                 raise ValueError('%s is not a GeometryEntity and cannot be made into Point' % str(e))
-
-    res = entities[0].intersection(entities[1])
-    for entity in entities[2:]:
-        newres = []
-        for x in res:
-            newres.extend(x.intersection(entity))
-        res = newres
-    return res
+    for j in range(0, len(entities)):
+        for k in range(j + 1, len(entities)):
+            if len(intersection(entities[j], entities[k])) is not 0:
```

Comment:
Thank You @smichr . I'm working on it.

_Sent from my LENOVO Lenovo P1a42 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_